src/CircleEIP1193Provider.ts

```diff
@@ -88,12 +88,25 @@ class CircleEIP1193Provider implements ethers.Provider {
       case 'eth_accounts':
         return Array.from(this.wallets.values());
       // Handle other methods as per your specific Circle API capabilities
+      case 'eth_call':
+        if (!args.params || args.params.length < 2) {
+          throw new Error('eth_call requires a transaction object and optionally a block tag');
+        }
+        return this.call(args.params[0] as TransactionRequest);
       default:
         throw new Error(`Method ${args.method} not supported by Circle API`);
     }
   }
 
   // Implemented EIP-1193 methods
+  call(tx: TransactionRequest, blockTag: string = 'latest'): Promise<string> { 
+    const res = this.queryContractState(tx.to as string, blockTag);
+    if (!res) {
+      throw new Error("call couldn\'t query contract state");
+    }
+    return res as Promise<string>;
+  }
+
```

Comment:
Rather take the arguments from `tx` and build the request you need for `to`, `blockchain`, `abiParameters`, `abiFunctionSignature`. Same for the response.